aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -33,100 +105,90 @@ SparseCsrTensor new_csr_tensor(const TensorOptions& options) {
       DispatchKeySet(dispatch_key), options.dtype());
 }
 
-// TODO: This constructor should probably use an ATen abstract method in order
-// to make autograd dispatch available for the CSR constructor. See the relevant
-// note in native_functions.yaml.
-Tensor sparse_csr_tensor(
-    const Tensor& crow_indices,
-    const Tensor& col_indices,
+Tensor _sparse_csr_tensor_unsafe(const Tensor& crow_indices, const Tensor& col_indices,
     const Tensor& values,
     IntArrayRef size,
     c10::optional<ScalarType> dtype,
     c10::optional<Layout> layout,
     c10::optional<Device> device,
     c10::optional<bool> pin_memory) {
-  // See [Note: hacky wrapper removal for TensorOptions]
+
   TensorOptions options = TensorOptions().dtype(dtype).layout(layout).device(device).pinned_memory(pin_memory);
+
+  TORCH_CHECK_NOT_IMPLEMENTED(
+    options.device().type() == kCPU || options.device().type() == kCUDA,
+     "Could not run '", "sparse_csr_tensor", "' from the '", options.device(), "' device.)");
```

Comment:
This check is redundant as `new_csr_tensor()` call does the same. Remove.

Reply:
Ok, I moved TORCH_CHECK_NOT_IMPLEMENTED to `new_csr_tensor` function.  Just for context TORCH_CHECK_NOT_IMPLEMENTED is important here, because when the `device` parameter provided in the constructor is  `meta` the tests can be skipped.